x-pack/legacy/plugins/uptime/server/rest_api/pings/get_all.ts

```diff
@@ -4,36 +4,45 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import Joi from 'joi';
-import { PingResults } from '../../../common/graphql/types';
+import { schema } from '@kbn/config-schema';
 import { UMServerLibs } from '../../lib/lib';
+import { UMRestApiRouteCreator } from '../types';
 
-export const createGetAllRoute = (libs: UMServerLibs) => ({
+export const createGetAllRoute: UMRestApiRouteCreator = (libs: UMServerLibs) => ({
   method: 'GET',
   path: '/api/uptime/pings',
+  validate: {
+    query: schema.object({
+      dateRangeStart: schema.string(),
+      dateRangeEnd: schema.string(),
```

Comment:
it went from Joi.number() to schema.string() , is this intentional?

Reply:
Yeah, that's the new method of specifying fields for the NP. There are examples in the [migration guide](https://github.com/elastic/kibana/blob/master/src/core/MIGRATION_EXAMPLES.md#route-registration).